nvflare/private/fed/app/simulator/simulator_runner.py

```diff
@@ -43,23 +44,45 @@
 
 
 class SimulatorRunner(FLComponent):
-    def __init__(self, args):
+    def __init__(self, job_folder: str, workspace: str, clients=None, n_clients=None, threads=None, gpu=None):
         super().__init__()
-        self.args = args
+
+        self.job_folder = job_folder
+        self.workspace = workspace
+        self.clients = clients
+        self.n_clients = n_clients
+        self.threads = threads
+        self.gpu = gpu
+
         self.ask_to_stop = False
+        self.args = None
 
         self.simulator_root = None
         self.services = None
         self.deployer = SimulatorDeployer()
         self.client_names = []
 
-        self.args.set = []
+    def _generate_args(self, job_folder: str, workspace: str, clients=None, n_clients=None, threads=None, gpu=None):
+        args = Namespace(
+            job_folder=job_folder,
+            workspace=workspace,
+            clients=clients,
+            n_clients=n_clients,
+            threads=threads,
+            gpu=gpu,
+        )
+        args.set = []
+        return args
 
     def setup(self):
-        if self.args.client_list:
-            self.client_names = self.args.client_list.strip().split(",")
-        elif self.args.clients:
-            for i in range(self.args.clients):
+        self.args = self._generate_args(
+            self.job_folder, self.workspace, self.clients, self.n_clients, self.threads, self.gpu
+        )
+
+        if self.args.clients:
+            self.client_names = self.args.clients.strip().split(",")
+        elif self.args.n_clients:
+            for i in range(self.args.n_clients):
                 self.client_names.append("site-" + str(i + 1))
```

Comment:
what happens if users specify both "clients" and "n_clients"? 
Should we give a warning or disallow that?

Reply:
"n_clients" will take precedence.

Reply:
actually "clients" will take precedence, but I get it, we just need to make sure we have noted this in doc.

Thanks